File: HyeonjeongKo/day2-49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = collections.defaultdict(list) # 빈 리스트

        # 각 단어를 정렬한값을 키로사용
        # 원래단어를 value로 사용
        for word in strs:
            # 파이썬은 list는 키값이될수없음.
            # 따라서 join을 사용해 str키값으로 반환
            anagrams[''.join(sorted(word))].append(word) 
        return list(anagrams.values())

File: JooEun/day2-49.cpp
class Solution {
public:
    // sort all the given elements and, group them when they have the same sorted string pattern
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> result; // 제출 vector
        
        // sort each string element to compare one another
        vector<string> sortedStr;
        for (string element: strs){
            string strCopy = element;
            sort(strCopy.begin(), strCopy.end()); // 정렬
            sortedStr.push_back(strCopy);
        }

        // brute force method => time limit 111/118 case passed
        // make string set that doesn't have duplication
        set<string> noDoubledStr;
        for(string ele : sortedStr){
            noDoubledStr.insert(ele);
        }

        // insert empty vector as many as the size of set that doesn't have duplication
        for(int i=0; i<noDoubledStr.size(); i++){
            vector<string> tmpArr;
            result.push_back(tmpArr);
        }
        
        for(int i=0; i<sortedStr.size(); i++){
            int j=0;
            set<string>::iterator iter;
            for(iter = noDoubledStr.begin(); iter != noDoubledStr.end(); iter++){
                if(sortedStr[i].compare(*iter)==0){
                    result[j].push_back(strs[i]);
                    break;
                }else{
                    j++;
                }
            }
        }

        
        return result;
    }
};
